github_connector/models/github_repository_branch.py

```diff
@@ -275,8 +293,78 @@ def _analyze_code(self):
                         branch.name,
                         e,
                     )
+        # github_analysis_rule
+        for branch in self:
+            rule_ids = self.env["github.analysis.rule"].search(
```

Comment:
Why doing this search instead of accessing directly to the rules?
```suggestion
            rule_ids = branch.repository_id.organization_id.analysis_rule_ids.mapped("analysis_rule_id") +                         branch.repository_id.analysis_rule_ids.mapped("analysis_rule_id")
```

Reply:
Sorry, i don't remember the motive about it, days ago code are very different and maybe before was neccesary but now not, it's changed with this suggested code.